src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,3222 @@
+    type Output BlockHeightInput = Proto.BlocksAtHeightRequest
+    toProto Relative {..} =
+        Proto.make $
+          ProtoFields.relative .= Proto.make ( do
+            ProtoFields.genesisIndex .= toProto rGenesisIndex
+            ProtoFields.height .= toProto rBlockHeight
+            ProtoFields.restrict .= rRestrict )
+    toProto Absolute {..} =
+        Proto.make $
+            ProtoFields.absolute .= Proto.make (ProtoFields.height .= toProto aBlockHeight)
+
+instance FromProto Proto.MintRate where
+    type Output' Proto.MintRate = MintRate
+    fromProto mr =
+        MintRate
+            { mrMantissa = mr ^. ProtoFields.mantissa
+            , mrExponent = fromIntegral $ mr ^. ProtoFields.exponent
+            }
+
+instance FromProto Proto.TokenomicsInfo where
+    type Output' Proto.TokenomicsInfo = QueryTypes.RewardStatus
+    fromProto ti =
+        case (ti ^? ProtoFields.v0, ti ^? ProtoFields.v1) of
+            (Just v0, Nothing) ->
+                QueryTypes.RewardStatusV0
+                    { rsTotalAmount = fromProto $ v0 ^. ProtoFields.totalAmount
+                    , rsTotalEncryptedAmount = fromProto $ v0 ^. ProtoFields.totalEncryptedAmount
+                    , rsBakingRewardAccount = fromProto $ v0 ^. ProtoFields.bakingRewardAccount
+                    , rsFinalizationRewardAccount = fromProto $ v0 ^. ProtoFields.finalizationRewardAccount
+                    , rsGasAccount = fromProto $ v0 ^. ProtoFields.gasAccount
+                    , rsProtocolVersion = fromProto $ v0 ^. ProtoFields.protocolVersion
+                    }
+            (Nothing, Just v1) ->
+                QueryTypes.RewardStatusV1
+                    { rsTotalAmount = fromProto $ v1 ^. ProtoFields.totalAmount
+                    , rsTotalEncryptedAmount = fromProto $ v1 ^. ProtoFields.totalEncryptedAmount
+                    , rsBakingRewardAccount = fromProto $ v1 ^. ProtoFields.bakingRewardAccount
+                    , rsFinalizationRewardAccount = fromProto $ v1 ^. ProtoFields.finalizationRewardAccount
+                    , rsGasAccount = fromProto $ v1 ^. ProtoFields.gasAccount
+                    , rsFoundationTransactionRewards = fromProto $ v1 ^. ProtoFields.foundationTransactionRewards
+                    , rsNextPaydayTime = snd . fromProto $ v1 ^. ProtoFields.nextPaydayTime
+                    , rsNextPaydayMintRate = fromProto $ v1 ^. ProtoFields.nextPaydayMintRate
+                    , rsTotalStakedCapital = fromProto $ v1 ^. ProtoFields.totalStakedCapital
+                    , rsProtocolVersion = fromProto $ v1 ^. ProtoFields.protocolVersion
+                    }
+            _ -> error "UwU"
+
+data InvokeInstanceInput = InvokeInstanceInput
+    { iiBlockHash :: BlockHashInput
+    , iiInvoker :: Maybe Address
+    , iiInstance :: ContractAddress
+    , iiAmount :: Amount
+    , iiEntrypoint :: Wasm.ReceiveName
+    , iiParameter :: Wasm.Parameter
+    , iiEnergy :: Energy
+    }
```

Comment:
This already exists as `data ContractContext = ContractContext`

Reply:
It is a little different. `ContractContext` has `Maybe BlockHash` rather than `BlockHashInput` . The protocol buffer schema also specifies a `BlockHashInput` in `InvokeInstanceRequest` which this type is modeled after.

If we use `ContractContext`, should this field simply not be set in the protocol buffer message when it is `Nothing`?